### DISCUSSION TITLE
how to use `vuejs3-logger` with quasar *the correct way*?

### DISCUSSION THREAD

**Discussion by s-light** [quasarframework/quasar#13531] [General - Components / Directives / etc] [answered]
i would like to use something like [`vuejs3-logger`](https://github.com/MarcSchaetz/vuejs3-logger) with my quasar app.

currently i feel a bit lost on how to setup this properly.
the vuejs3-logger documentation tells me to *setup* it this way:
```js
import VueLogger from 'vuejs3-logger';
import { createApp } from 'vue';

const isProduction = process.env.NODE_ENV === 'production';
 
const options = {
    isEnabled: true,
    logLevel : isProduction ? 'error' : 'debug',
    stringifyArguments : false,
    showLogLevel : true,
    showMethodName : true,
    separator: '|',
    showConsoleColors: true
};

createApp({}).use(VueLogger, options);
```

so i translated this to a boot file:
```js
import { boot } from 'quasar/wrappers'

import VueLogger from 'vuejs3-logger';

export default boot(({app}) => {
  const isProduction = process.env.NODE_ENV === 'production';
  const options = {
      // https://github.com/MarcSchaetz/vuejs3-logger#properties
      isEnabled: true,
      logLevel : isProduction ? 'error' : 'debug',
      stringifyArguments : false,
      showLogLevel : true,
      showMethodName : true,
      separator: '|',
      showConsoleColors: true
  };
  app.use(VueLogger, options);
})
```

in my `HelloWorldPage.vue` file i have 
```vue
<template>
  <q-page class="flex flex-center">
      <q-btn label="log something.." @click="logSomething"/>
  </q-page>
</template>

<script setup>
import { defineComponent, inject } from 'vue'

const log = inject('vuejs3-logger')
log.debug('test', 42)

const logSomething = (event) => {
    log.debug('logSomething called..', event.target)
    console.log('logSomething called..', event.target)
}
</script>
```

in the console i get
> 18:08:44.992 debug |  /node_modules/@vue/runtime-core/dist/runtime-core |  test 42       [vue3-logger.js:99](webpack://xxx/node_modules/vuejs3-logger/dist/vue3-logger.js?fefa)
> 18:08:47.020 debug |  /node_modules/@vue/runtime-core/dist/runtime-core |  logSomething called..  `<span class="block">`       [vue3-logger.js:99](webpack://xxx/node_modules/vuejs3-logger/dist/vue3-logger.js?fefa)
> 18:08:47.021 logSomething called..  `<span class="block">`          [HelloWorldPage.vue:15](webpack://xxx/src/pages/HelloWorldPage.vue?eb2f)

so basically it is working -
but the print function output is a bit off..
i think that is doe to how quasar is setup...
or how i setup things?

if anybody has any tips or other tools to do some sort of logging in quasar please let me know :-)

**Comment by metalsadman** [comment, marked as answer]
the boot file and the logs looks fine to me, I don't think that's a quasar issue, more like a verbosity config on that library that you're using.

**Reply by s-light** [reply]
ok thanks!
i have [posted this as issue](https://github.com/MarcSchaetz/vuejs3-logger/issues/4) at the library repository

**Reply by cdecinkoKnight** [reply]
Has there been any update? Is this a good logging library to use?

**Reply by metalsadman** [reply]
you can bake your own around the console api imho.
